port/cpl_vsi_mem.cpp

```diff
@@ -1184,3 +1342,43 @@ GByte *VSIGetMemFileBuffer(const char *pszFilename, vsi_l_offset *pnDataLength,
 
     return pabyData;
 }
+
+/************************************************************************/
+/*                   VSIMemGenerateUnlistedFilename()                   */
+/************************************************************************/
+
+/**
+ * \brief Generates a unique filename that can be used with the /vsimem/
+ * virtual file system.
+ *
+ * This function returns a (short-lived) string containing a unique filename,
```

Comment:
Would it make sense to return a `CPLString` ? I'm having trouble understanding under what circumstance I could use the "short-lived" object.

Reply:
if we exposed only in the C++ API, yes, but I also wanted it to be available for the C API.
Returning short-lived strings is common in the GDAL file API. ALL functions like CPLGetPath(), CPLGetExtension(), etc. do that.